src/main/java/org/opensearch/neuralsearch/search/query/HybridCollectorManager.java

```diff
@@ -149,9 +190,83 @@ public ReduceableSearchResult reduce(Collection<Collector> collectors) {
             TopDocsAndMaxScore topDocsAndMaxScore = new TopDocsAndMaxScore(newTopDocs, hybridTopScoreDocCollector.getMaxScore());
             return (QuerySearchResult result) -> { result.topDocs(topDocsAndMaxScore, getSortValueFormats(sortAndFormats)); };
         }
+
+        if (!hybridSortedTopDocCollectors.isEmpty()) {
+            HybridTopFieldDocSortCollector hybridSortedTopScoreDocCollector = hybridSortedTopDocCollectors.stream()
+                .findFirst()
+                .orElseThrow(() -> new IllegalStateException("cannot collect results of hybrid search query"));
+
+            SimpleFieldCollector simpleFieldCollector;
+            PagingFieldCollector pagingFieldCollector;
+            List<TopFieldDocs> topFieldDocs;
+            long maxTotalHits;
+            float maxScore;
+            if (hybridSortedTopScoreDocCollector instanceof SimpleFieldCollector) {
+                simpleFieldCollector = (SimpleFieldCollector) hybridSortedTopScoreDocCollector;
+                topFieldDocs = simpleFieldCollector.topDocs();
+                maxTotalHits = simpleFieldCollector.getTotalHits();
+                maxScore = simpleFieldCollector.getMaxScore();
+            } else {
+                pagingFieldCollector = (PagingFieldCollector) hybridSortedTopScoreDocCollector;
+                topFieldDocs = pagingFieldCollector.topDocs();
+                maxTotalHits = pagingFieldCollector.getTotalHits();
+                maxScore = pagingFieldCollector.getMaxScore();
+            }
+
+            TopDocs newTopDocs = getNewTopFieldDocs(
+                getTotalHits(this.trackTotalHitsUpTo, topFieldDocs, isSingleShard, maxTotalHits),
+                topFieldDocs,
+                sortAndFormats.sort.getSort()
+            );
+            TopDocsAndMaxScore topDocsAndMaxScore = new TopDocsAndMaxScore(newTopDocs, maxScore);
+            return (QuerySearchResult result) -> { result.topDocs(topDocsAndMaxScore, getSortValueFormats(sortAndFormats)); };
```

Comment:
lets move this to a private function in this class. We can move the above if condition but that you can take iteratively.